tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -496,11 +496,13 @@ private:
          RIterator(const RCollectionIterableOnce &owner) : fOwner(owner) {}
          RIterator(const RCollectionIterableOnce &owner, void *iter) : fOwner(owner), fIterator(iter)
          {
-            fElementPtr = fOwner.fIFuncs.fNext(fIterator, fOwner.fEnd);
+            fElementPtr =
+               fOwner.fStride ? *static_cast<void **>(fIterator) : fOwner.fIFuncs.fNext(fIterator, fOwner.fEnd);
```

Comment:
This feels odd.   Previously implementation had that code identical to `operator++`.   

It seems that either the code is expecting the creation of the `RIterator` to move the cursor (in which case the 'true' branch is missing an increment) or it does not (in which case the `false` branch is leading to the first element being skipped).  Either way there would be a problem here (and almost as importantly in the test that did not point to this issue).

Alternatively, what am I missing?

Reply:
The `RIterator` constructor initializes the iterator to point to the first element; however, in that case, the side effect of the `fNext()` function is to increment the iterator.  It will, and actually did, work as long as the initialization, `operator++` and "end condition" is consistent across the same case (stride known / unknown).

However, I do totally agree that it looks inconsistent :+1:.  Thus, I have updated the code to fix that.

Reply:
Do we have a test that fails because of the wrong code?  Shouldn't we add one if not?

Reply:
Yes :slightly_smiling_face:, we are already testing that [here](https://github.com/root-project/root/blob/master/tree/ntuple/v7/test/ntuple_types.cxx#L1052).  Specifically, we test for `kSTLvector` (which assumes the elements are laid out contiguously), and `kSTLdeque` (which issues the indirect call to `fNext()` instead).

Reply:
Good.  So those test failed with the previous code?